electrum/lnchannel.py

```diff
@@ -1513,19 +1513,25 @@ def consider_ctx(*, ctx_owner: HTLCOwner, is_htlc_dust: bool) -> int:
             # handle the transaction fees for the HTLC transaction
             if is_htlc_dust:
                 # nobody pays additional HTLC transaction fees
-                return min(max_send_msat, htlc_trim_threshold_msat - 1)
+                max_send_msat = min(max_send_msat, htlc_trim_threshold_msat - 1)
             else:
                 # somebody has to pay for the additonal HTLC transaction fees
                 if sender == initiator:
-                    return max_send_msat - htlc_fee_msat
+                    max_send_msat = max_send_msat - htlc_fee_msat
                 else:
                     # check if the receiver can afford to pay for the HTLC transaction fees
                     new_receiver_balance = receiver_balance_msat - receiver_reserve_msat - ctx_fees_msat[receiver] - htlc_fee_msat
                     if self.has_anchors():
                         new_receiver_balance -= 2 * FIXED_ANCHOR_SAT * 1000
                     if new_receiver_balance < 0:
-                        return 0
-                    return max_send_msat
+                        max_send_msat = 0
+
+            if single_payment:  # check max_htlc_value_in_flight_msat
+                current_htlc_sum = htlcsum(self.hm.htlcs_by_direction(receiver, direction=RECEIVED, ctn=ctn).values())
+                remaining_max_inflight = self.config[receiver].max_htlc_value_in_flight_msat - current_htlc_sum
+                max_send_msat = min(max_send_msat, remaining_max_inflight)
+
```

Comment:
why is this inside the consider_ctx function?

Reply:
indeed, moved.